ksqldb-functional-tests/src/test/resources/query-validation-tests/fk-join.json

```diff
@@ -1,5 +1,35 @@
 {
   "tests": [
+    {
+      "name": "Should fail if single result key column is missing for inner-join",
```

Comment:
new test for the new check

Reply:
As discussed offline, it'd be great to get https://github.com/confluentinc/ksql/pull/7537 merged first so this PR can simply uncomment the tests that are now expected to pass, rather than adding any new tests and requiring us to resolve merge conflicts / the possibility of duplicate tests.